### PR TITLE
ENDOC-417b Refine Entando Identity Management -- Keycloak

### PR DESCRIPTION
**Pull request by Lyd1aCla1r3**
second pr to add changes to both next and 6.3.2 within new nav

### CODE REVIEW

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 1]
```diff
@@ -1,53 +1,67 @@
 # Entando Identity Management -- Keycloak
 
-Entando Identity Management is powered by Keycloak. The Keycloak instance used for your Entando apps
-can be [externally installed](../../tutorials/devops/external-id-management) or you can use a dedicated instance on a per application basis. The sections below details the architecture and documentation required to customize your Keycloak instance.
+Entando Identity Management is based on open source Keycloak. Entando Applications rely on a Keycloak instance that is either[externally installed](../../tutorials/devops/external-id-management) or specific to an application. The architecture and requirements to customize your Keycloak instance are described below.
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/identity-management.md:3]: There's a missing space after "either".

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 2]
```diff
@@ -1,53 +1,67 @@
...
 
 ### Plugins/Microservices
-Authorization for the microservices comes from clients and roles in keycloak. The authorizations are stored in the JWT token and are available to the services when invoked.
+Keycloak authorizes microservices using clients and roles. Authorizations are stored in a JWT token and available to services when invoked.
 
-### Core
-Authorization for the entando-core and WCMS is provided by the Entando user management database. When a user is authenticated to the entando-core  a copy of that user is added to the Entando user management database.  That copy is made in support of the authorization flow noted below.
+To setup permissions to configure the service:
+
+1. Login to your Keycloak instance as an admin. You can query Kubernetes for the default admin credentials, substituting your environment's namespace and Secret name:
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/identity-management.md:33]: I think this step duplicates the first section on the page. Can we simplify by linking to #logging-into-your-keycloak-instance?

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 3]
```diff
@@ -1,53 +1,67 @@
...
 
 ## Authorization
 
 ### Plugins/Microservices
-Authorization for the microservices comes from clients and roles in keycloak. The authorizations are stored in the JWT token and are available to the services when invoked.
+Keycloak authorizes microservices using clients and roles. Authorizations are stored in a JWT token and available to services when invoked.
 
-### Core
-Authorization for the entando-core and WCMS is provided by the Entando user management database. When a user is authenticated to the entando-core  a copy of that user is added to the Entando user management database.  That copy is made in support of the authorization flow noted below.
+To setup permissions to configure the service:
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/identity-management.md:31]: "These are the steps to grant a user one or more roles for a specific client. Note: when a microservice is installed in Entando, a corresponding client (and set of roles) is created per its [plugin definition](docs/ecr/ecr-bundle-details.html#plugin) "  

@Lyd1aCla1r3 I'm pretty sure my link is wrong but you get the idea...

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 4]
```diff
@@ -1,53 +1,67 @@
...
+
+1. Login to your Keycloak instance as an admin. You can query Kubernetes for the default admin credentials, substituting your environment's namespace and Secret name:
+
+```
+ent kubectl get secret quickstart-kc-admin-secret -n entando -o go-template="{{println}}Username: {{.data.username | base64decode}}{{println}}Password: {{.data.password | base64decode}}{{println}}{{println}}"
+```
+
+2. From the left menu, select `Users`. Search for "admin" and click on the admin ID.
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/identity-management.md:39]: "Go to `Users` and find the appropriate user, e.g. "admin""

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 5]
```diff
@@ -1,53 +1,67 @@
...
 
-As noted above when a user is authenticated to the entando-core via keycloak a copy of that user is added to the entando-core user management database in support of WCMS functionality.  Using the App Builder WCMS roles and groups can be assigned to a user for access to functions in the App Builder or for portal-ui based content access in the runtime application The code that copies the user into the entando-core can be customized to automatically create groups and roles as needed for an application but it is something that must be done on a per implementation basis.
+![find-admin.png](./img/find-admin.png)
 
-For more details on the code that copies users and data to the WCMS database see the [entando-keycloak-plugin](https://github.com/entando/entando-keycloak-plugin). The readme in that project includes properties that are available to your Entando app.
+3. Click on the `Role Mappings` tab. Specify the service client via the `Client Roles` drop-down menu. Select from the client's `Available Roles`.
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/identity-management.md:43]: This is 3 steps. Break it up for clarity?

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 6]
```diff
@@ -1,53 +1,67 @@
...
 
-For a deeper look See: [KeycloakAuthorizationManager.java](https://github.com/entando/entando-keycloak-plugin/blob/master/src/main/java/org/entando/entando/keycloak/services/KeycloakAuthorizationManager.java) in the plugin for an example of adding attributes programatically. In particular, the [processNewUser](https://github.com/entando/entando-keycloak-plugin/blob/master/src/main/java/org/entando/entando/keycloak/services/KeycloakAuthorizationManager.java#L43) method.
+![find-roles.png](./img/find-roles.png)
+
+4. Use the `Add Selected` button to move desired roles to `Assigned Roles`. These will subsequently appear under `Effective Roles`.
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/identity-management.md:47]: "the desired roles"

**Review by nshaw — CHANGES_REQUESTED** [review]

File: vuepress/docs/next/docs/consume/identity-management.md (hunk 2 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/consume/identity-management.md:33]: i feel the same and tried to do that but couldn't! let me try again

**Comment by Lyd1aCla1r3** [comment]
@nshaw so the thinking was to combine steps so i could show multiple things with one pic and avoid being inconsistent with steps and pics - like some steps have associated pics and some don't. but, it was really hard to do that well and i struggled with it. can we split the steps up into single actions but include pics sparingly? as in increase steps but keep the number of pics the same?

File: vuepress/docs/next/docs/consume/identity-management.md (hunk 3 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/consume/identity-management.md:31]: should the link go here: https://developer.entando.com/next/tutorials/backend-developers/generate-microservices-and-micro-frontends.html#overview or are we talking a repo of yaml files? i know better than to link to GitHub without making sure it's front-facing and not subject to change with revs...

Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/consume/identity-management.md:31]: sorry, @nshaw

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 7]
```diff
@@ -1,53 +1,66 @@
...
+and search for the Secret that ends in `kc-admin-secret`.
 
 ## Authentication
-In Entando 6 all authentication goes through Keycloak. This ensures that a micro frontend can call a microservice with a token that is available on the client.
+Beginning with Entando 6, all authentication is powered by Keycloak. This ensures that a micro frontend can call a microservice with a token available to the client.
 
 ![Init Containers Screenshot](./img/keycloak-arch-high-level.png)
 
-By using Keycloak as a central point of authentication the Entando architecture is able to provide a single unified view of identity to the entire architecture. With this architecture Entando becomes more portable and can be integrated into other IDPs without changes to the source. Keycloak acts as an  abstraction to the underlying IDP.
+The Entando architecture implements Keycloak as a central point of authentication. A single, unified view of identity increases Entando's portability. Entando can also integrate into other IDPs without modifying the source, because Keycloak acts as an abstraction to the underlying IDP.
```
Comment (jyunmitch) [inline vuepress/docs/next/docs/consume/identity-management.md:24]: Does the Entando architecture implement Keycloak or does it provide a unified view of id... >> the original verbiage gave a clearer view. This seemed more confusing. What is IDP?

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 8]
```diff
@@ -1,53 +1,66 @@
...
 
 ## Authorization
 
 ### Plugins/Microservices
-Authorization for the microservices comes from clients and roles in keycloak. The authorizations are stored in the JWT token and are available to the services when invoked.
+Keycloak authorizes microservices using clients and roles. Authorizations are stored in a JWT token and available to services when invoked.
```
Comment (jyunmitch) [inline vuepress/docs/next/docs/consume/identity-management.md:29]: what is JWT?

File: vuepress/docs/next/docs/consume/identity-management.md [hunk 9]
```diff
@@ -1,53 +1,66 @@
...
+8. Use the `Add Selected` button to move the desired roles to `Assigned Roles`. These will subsequently appear under `Effective Roles`.
 
-For a deeper look See: [KeycloakAuthorizationManager.java](https://github.com/entando/entando-keycloak-plugin/blob/master/src/main/java/org/entando/entando/keycloak/services/KeycloakAuthorizationManager.java) in the plugin for an example of adding attributes programatically. In particular, the [processNewUser](https://github.com/entando/entando-keycloak-plugin/blob/master/src/main/java/org/entando/entando/keycloak/services/KeycloakAuthorizationManager.java#L43) method.
+![assign-roles.png](./img/assign-roles.png)
+### Core
+When a user is authenticated to the `entando-core` via Keycloak, a copy of that user is added to the `entando-core` user management database to enable WCMS functionality. Within the App Builder, WCMS roles and groups can be assigned to a user for access to App Builder functions or `portal-ui` content in the runtime application.
```
Comment (jyunmitch) [inline vuepress/docs/next/docs/consume/identity-management.md:50]: "copy of that user" >> should it say copy of that user information or something like that?

"runtime application" >> is this referring to specific types of programs, I think called runtime programs, or is it talking about the execution of an application?

File: vuepress/docs/next/docs/consume/identity-management.md (hunk 7 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/consume/identity-management.md:24]: the architecture implements Keycloak to provide a unified view. it should be obvious that central point of authentication provides a single, unified view ?? will rework...

File: vuepress/docs/next/docs/consume/identity-management.md (hunk 9 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/consume/identity-management.md:50]: no, copy of the user -> copy of the entity. similar to "copy the file" and not "copy the file's information"

leaving the runtime language as-is; just from a google search "runtime application" seems to be used frequently and sometimes interchangeably with runtime program. per one definition:
"A runtime program is a specific version of an application that is distributed for limited use. These kinds of software releases may be called runtime programs because they offer a runtime environment without other features that allow for more permanent use, such as building executable files."
given this def application seems more appropriate than program ?? i suspect there are nuances... @nshaw

Comment (nshaw) [reply vuepress/docs/next/docs/consume/identity-management.md:50]: 1. Copy of the user is good enough. It's really only a couple fields but functionally that's how it behaves.
2. Yes, application is better. It's somewhat interchangeable but we build web applications not web programs. Programs just sounds dated at this point for some reason.

**Review by nshaw — APPROVED** [review]

File: vuepress/docs/next/docs/consume/identity-management.md (hunk 9 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/consume/identity-management.md:50]: I must have worded this badly. I wasn't suggesting we change application to program. It was unclear whether you were talking about the application during the runtime or something else.

**Review by jyunmitch — APPROVED** [review]
